src/map/utils/battleutils.cpp

```diff
@@ -730,6 +730,16 @@ namespace battleutils
 
                         if (PDefender->isAlive())
                         {
+                            // Subpower is the remaining damage that can be drained. When it reaches 0 the effect ends
+                            int remainingDrain = PDefender->StatusEffectContainer->GetStatusEffect(EFFECT_DREAD_SPIKES)->GetSubPower();

```

Comment:
Wait is there a way to make it grab whatever spikes effect you have? Otherwise I'd have to grab the effect in each case individually which... doesn't make it usable in both places and would be confusing.